server/issue.go

```diff
@@ -492,50 +480,28 @@ func httpAPIAttachCommentToIssue(ji Instance, w http.ResponseWriter, r *http.Req
 			errors.WithMessage(err, "failed to attach the comment, postId: "+attach.PostId)
 	}
 
-	if len(post.FileIds) > 0 {
-		go func() {
-			updated := false
-
-			// Upload the files
-			for _, fileId := range post.FileIds {
-				info, ae := api.GetFileInfo(fileId)
-				if ae != nil {
-					continue
-				}
-				// TODO: large file support? Ignoring errors for now is good enough...
-				byteData, ae := api.ReadFile(info.Path)
-				if ae != nil {
-					notifyOnFailedAttachment(ji, ae.Error(), mattermostUserId, info.Path, attach.IssueKey)
-					continue
-				}
-
-				attachments, _, err2 := jiraClient.Issue.PostAttachment(attach.IssueKey, bytes.NewReader(byteData), info.Name)
-
-				if err2 != nil {
-					notifyOnFailedAttachment(ji, err2.Error(), mattermostUserId, info.Path, attach.IssueKey)
-					continue
-				}
-				if attachments != nil {
-					// There will only ever be one attachment at a time.
-					attachment := (*attachments)[0]
-					jiraComment.Body += "\n\nAttachment: !" + attachment.Filename + "!"
-					updated = true
-				}
+	go func() {
```

Comment:
Aside from the `extraText` variable, this anonymous function could be named, refactored to shared with the code above.  https://github.com/mattermost/mattermost-plugin-jira/blob/e53571495492ff59f2fa43bfa6a8b61dd2ed9f01/server/issue.go#L187-L195

Reply:
0/5 because there are larger swaths of code to focus on during refactoring.

Reply:
That's how I first did it, but I had to pass all the extra parameters back and forth, and it was ugly. I figured, `attachFileToIssue` was good enough a refactor, simple and explicit.